subsys/storage/flash_buffered_write/fbw.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2017, 2020 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME FBW
+#define LOG_LEVEL CONFIG_FBW_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_FBW_LOG_LEVEL);
+
+#include <zephyr/types.h>
+#include <string.h>
+#include <drivers/flash.h>
+
+#include <storage/fbw.h>
+
+#ifdef CONFIG_FBW_ERASE
+
+int fbw_erase(struct fbw_ctx *ctx, off_t off)
+{
+	int rc;
+	struct flash_pages_info page;
+
+	rc = flash_get_page_info_by_offs(ctx->fdev, off, &page);
+	if (rc) {
+		LOG_ERR("Error %d while getting page info", rc);
+		return rc;
+	}
+
+	if (ctx->last_erased_page_start_offset == page.start_offset) {
+		return 0;
+	}
+
+	ctx->last_erased_page_start_offset = page.start_offset;
+	LOG_INF("Erasing page at offset 0x%08lx", (long)page.start_offset);
+
+	flash_write_protection_set(ctx->fdev, false);
+	rc = flash_erase(ctx->fdev, page.start_offset, page.size);
+	flash_write_protection_set(ctx->fdev, true);
+
+	if (rc) {
+		LOG_ERR("Error %d while erasing page", rc);
+	}
+
+	return rc;
+}
+
+#endif /* CONFIG_FBW_ERASE */
+
+static int flash_sync(struct fbw_ctx *ctx)
+{
+	int rc = 0;
+	size_t write_addr = ctx->offset + ctx->bytes_written;
+
+
+	if (IS_ENABLED(CONFIG_FBW_ERASE)) {
+		rc = fbw_erase(ctx, write_addr + ctx->buf_bytes);
+		if (rc) {
+			LOG_ERR("fbw_erase error %d offset=0x%08zx", rc,
+					write_addr);
+			return rc;
+		}
+	}
+
+	flash_write_protection_set(ctx->fdev, false);
+	rc = flash_write(ctx->fdev, write_addr, ctx->buf, ctx->buf_bytes);
+	flash_write_protection_set(ctx->fdev, true);
+
+	if (rc) {
```

Comment:
not consistent, below it is `if (rc != 0) {`

Reply:
Changed to `if (rc)` everywhere.